Add more intuitive exception message when invoked destroy on cache being used by other clients

Client version:
```ehcache-clustered
Implementation-Version: 3.1.0-SNAPSHOT 3d44aa4c4a3047b773beab6ee761262
 036491394
```

Server version:
```
Terracotta 5.0.0-beta2, as of 2016-06-30 at 01:31:45 PDT (Revision 94748627d9b6fb3e06e6ccad724f821ef1ef741d from UNKNOWN)
```

When called **cacheManager.destroyCache("clustered-cache")** from one client while there are other clients still using the "clustered-cache", below exception is hit. 

@AbfrmBlr mentioned this is the expected behavior as unless all the other clients using this cache are terminated and then the destroyCache called from a single client, this is expected to work.
If that is the case, then the exception message should be more explicitly saying this.

```
java.lang.AssertionError: Caught exception during execution
Expected: is null
     but: was <org.ehcache.StateTransitionException: Couldn't acquire cluster-wide maintenance lease>

	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at com.terracotta.qa.ehcache.test.MultipleClientsTest.testCRUDOps(MultipleClientsTest.java:172)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:200)
	at org.jboss.byteman.contrib.bmunit.BMNGAbstractRunner.run(BMNGAbstractRunner.java:58)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:689)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:74)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:121)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
```

@ljacomet - complete trace as below.
Expectation is to provide clear message to end user to invoke destroyCache only after all the other clients are closed.  Perhaps, complete trace is indirectly stating the same? if yes, no changes might be needed here.

````
org.ehcache.CachePersistenceException: Cannot destroy clustered tier 'clustered-cache' on terracotta://localhost:9510
	at org.ehcache.clustered.client.internal.service.DefaultClusteringService.destroy(DefaultClusteringService.java:313)
	at org.ehcache.core.EhcacheManager.destroyPersistenceSpace(EhcacheManager.java:666)
	at org.ehcache.core.EhcacheManager.destroyCache(EhcacheManager.java:659)
	at com.terracotta.qa.ehcache.test.MultipleClientsTest.dummyTest(MultipleClientsTest.java:291)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:85)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:200)
	at org.jboss.byteman.contrib.bmunit.BMNGAbstractRunner.run(BMNGAbstractRunner.java:58)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:212)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:689)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:882)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1189)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:124)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:108)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:348)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:343)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:305)
	at org.testng.SuiteRunner.run(SuiteRunner.java:254)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1224)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1149)
	at org.testng.TestNG.run(TestNG.java:1057)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:74)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:121)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: org.ehcache.clustered.client.internal.service.ClusteredTierDestructionException: Error destroying clustered tier 'clustered-cache'
	at org.ehcache.clustered.client.internal.EhcacheClientEntity.destroyCache(EhcacheClientEntity.java:206)
	at org.ehcache.clustered.client.internal.service.DefaultClusteringService.destroy(DefaultClusteringService.java:311)
	... 34 more
Caused by: org.ehcache.clustered.common.internal.exceptions.ResourceBusyException: Can not destroy clustered tier 'clustered-cache': in use by 1 clients
	at org.ehcache.clustered.common.internal.exceptions.ResourceBusyException.withClientStackTrace(ResourceBusyException.java:43)
	at org.ehcache.clustered.common.internal.exceptions.ResourceBusyException.withClientStackTrace(ResourceBusyException.java:22)
	at org.ehcache.clustered.common.internal.messages.ResponseCodec.decode(ResponseCodec.java:117)
	at org.ehcache.clustered.common.internal.messages.EhcacheCodec.decodeResponse(EhcacheCodec.java:67)
	at org.ehcache.clustered.common.internal.messages.EhcacheCodec.decodeResponse(EhcacheCodec.java:22)
	at com.tc.object.EntityClientEndpointImpl$InvocationBuilderImpl$1.getWithTimeout(EntityClientEndpointImpl.java:174)
	at com.tc.object.EntityClientEndpointImpl$InvocationBuilderImpl$1.getWithTimeout(EntityClientEndpointImpl.java:156)
	at org.ehcache.clustered.client.internal.EhcacheClientEntity.waitFor(EhcacheClientEntity.java:276)
	at org.ehcache.clustered.client.internal.EhcacheClientEntity.invokeInternal(EhcacheClientEntity.java:240)
	at org.ehcache.clustered.client.internal.EhcacheClientEntity.destroyCache(EhcacheClientEntity.java:204)
	... 35 more
Caused by: org.ehcache.clustered.common.internal.exceptions.ResourceBusyException: Can not destroy clustered tier 'clustered-cache': in use by 1 clients
	at org.ehcache.clustered.server.EhcacheActiveEntity.destroyServerStore(EhcacheActiveEntity.java:842)
	at org.ehcache.clustered.server.EhcacheActiveEntity.invokeLifeCycleOperation(EhcacheActiveEntity.java:359)
	at org.ehcache.clustered.server.EhcacheActiveEntity.invoke(EhcacheActiveEntity.java:307)
	at org.ehcache.clustered.server.EhcacheActiveEntity.invoke(EhcacheActiveEntity.java:89)
	at com.tc.objectserver.entity.ManagedEntityImpl.lambda$performAction$38(ManagedEntityImpl.java:610)
	at com.tc.objectserver.entity.ManagedEntityImpl.runWithHelper(ManagedEntityImpl.java:259)
	at com.tc.objectserver.entity.ManagedEntityImpl.performAction(ManagedEntityImpl.java:610)
	at com.tc.objectserver.entity.ManagedEntityImpl.invoke(ManagedEntityImpl.java:358)
	at com.tc.objectserver.entity.ManagedEntityImpl.lambda$processInvokeRequest$31(ManagedEntityImpl.java:204)
	at com.tc.objectserver.entity.ManagedEntityImpl$SchedulingRunnable.run(ManagedEntityImpl.java:837)
	at com.tc.objectserver.entity.RequestProcessor$EntityRequest.invoke(RequestProcessor.java:219)
	at com.tc.objectserver.entity.RequestProcessor$EntityRequest.run(RequestProcessor.java:211)
	at com.tc.objectserver.entity.RequestProcessorHandler.handleEvent(RequestProcessorHandler.java:27)
	at com.tc.objectserver.entity.RequestProcessorHandler.handleEvent(RequestProcessorHandler.java:23)
	at com.tc.async.impl.StageQueueImpl$HandledContext.runWithHandler(StageQueueImpl.java:502)
	at com.tc.async.impl.StageImpl$WorkerThread.run(StageImpl.java:197)

````

--- a/clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java
+++ b/clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java
@@ -27,6 +27,7 @@ import org.ehcache.clustered.common.internal.ClusteredEhcacheIdentity;
 import org.ehcache.clustered.common.ServerSideConfiguration;
 import org.ehcache.clustered.common.internal.ServerStoreConfiguration;
 import org.ehcache.clustered.common.internal.exceptions.ClusterException;
+import org.ehcache.clustered.common.internal.exceptions.ResourceBusyException;
 import org.ehcache.clustered.common.internal.messages.EhcacheEntityMessage;
 import org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse;
 import org.ehcache.clustered.common.internal.messages.EhcacheEntityResponse.Failure;
@@ -202,6 +203,8 @@ public class EhcacheClientEntity implements Entity {
   public void destroyCache(String name) throws ClusteredTierDestructionException, TimeoutException {
     try {
       invokeInternal(timeouts.getLifecycleOperationTimeout(), messageFactory.destroyServerStore(name), true);
+    } catch (ResourceBusyException e) {
+      throw new ClusteredTierDestructionException(e.getMessage(), e);
     } catch (ClusterException e) {
       throw new ClusteredTierDestructionException("Error destroying clustered tier '" + name + "'", e);
     }

--- a/clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java
+++ b/clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/DefaultClusteringService.java
@@ -310,7 +310,7 @@ class DefaultClusteringService implements ClusteringService, EntityService {
     try {
       entity.destroyCache(name);
     } catch (ClusteredTierDestructionException e) {
-      throw new CachePersistenceException("Cannot destroy clustered tier '" + name + "' on " + clusterUri, e);
+      throw new CachePersistenceException(e.getMessage() + " (on " + clusterUri + ")", e);
     } catch (TimeoutException e) {
       throw new CachePersistenceException("Could not destroy clustered tier '" + name + "' on " + clusterUri
           + "; destroy operation timed out" + clusterUri, e);

--- a/clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java
+++ b/clustered/client/src/test/java/org/ehcache/clustered/client/CacheManagerDestroyTest.java
@@ -36,7 +36,6 @@ import java.net.URI;
 import static org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder.cluster;
 import static org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder;
 import static org.hamcrest.Matchers.is;
-import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.fail;
 
@@ -168,38 +167,6 @@ public class CacheManagerDestroyTest {
     Cache<Long, String> cache2 = persistentCacheManager2.getCache("test", Long.class, String.class);
 
     assertThat(cache2.get(1L), is("One"));
-
-  }
-
-  @Test
-  public void testDestroyUnKnownCacheAlias() throws CachePersistenceException {
-    CacheManagerBuilder<PersistentCacheManager> cacheManagerBuilder = clusteredCacheManagerBuilder
-        .withCache("test", CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
-            ResourcePoolsBuilder.newResourcePoolsBuilder()
-                .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 32, MemoryUnit.MB))));
-
-    PersistentCacheManager persistentCacheManager1 = cacheManagerBuilder.build(true);
-
-    PersistentCacheManager anotherPersistentCacheManager = clusteredCacheManagerBuilder.build(true);
-
-    try {
-      anotherPersistentCacheManager.destroyCache("test");
-      fail("CachePersistenceException Expected");
-    } catch (CachePersistenceException e) {
-      assertThat(e.getMessage(), is("Cannot destroy clustered tier 'test' on terracotta://example.com:9540"));
-    }
-
-    persistentCacheManager1.close();
-
-    anotherPersistentCacheManager.destroyCache("test");
-
-    //Proves that the resource cache is destroyed.
-
-    Cache<Long, String> cache1 = anotherPersistentCacheManager.createCache("test", CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class,
-        ResourcePoolsBuilder.newResourcePoolsBuilder()
-            .with(ClusteredResourcePoolBuilder.clusteredDedicated("primary-server-resource", 40, MemoryUnit.MB))));
-
-    assertNotNull(cache1);
 
   }
 

--- a/clustered/client/src/test/java/org/ehcache/clustered/client/ClusteredCacheDestroyTest.java
+++ b/clustered/client/src/test/java/org/ehcache/clustered/client/ClusteredCacheDestroyTest.java
@@ -34,9 +34,11 @@ import org.junit.Test;
 
 import java.net.URI;
 
+import static org.ehcache.clustered.client.config.builders.ClusteredResourcePoolBuilder.clustered;
 import static org.ehcache.clustered.client.config.builders.ClusteringServiceConfigurationBuilder.cluster;
 import static org.ehcache.config.builders.CacheConfigurationBuilder.newCacheConfigurationBuilder;
 import static org.ehcache.config.builders.CacheManagerBuilder.newCacheManagerBuilder;
+import static org.ehcache.config.builders.ResourcePoolsBuilder.newResourcePoolsBuilder;
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
@@ -104,6 +106,23 @@ public class ClusteredCacheDestroyTest {
 
     anotherCache.put(1L, "One");
     assertThat(anotherCache.get(1L), is("One"));
+  }
+
+  @Test
+  public void testDestroyUnknownCacheAlias() throws Exception {
+    clusteredCacheManagerBuilder.build(true).close();
+
+    PersistentCacheManager cacheManager = newCacheManagerBuilder().with(cluster(CLUSTER_URI).expecting()).build(true);
+
+    cacheManager.destroyCache("clustered-cache");
+
+    try {
+      cacheManager.createCache("clustered-cache", newCacheConfigurationBuilder(Long.class, String.class, newResourcePoolsBuilder()
+          .with(clustered())));
+      fail("Expected exception as clustered store no longer exists");
+    } catch (IllegalStateException e) {
+      assertThat(e.getMessage(), containsString("clustered-cache"));
+    }
   }
 
   @Test

--- a/clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java
+++ b/clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java
@@ -691,7 +691,7 @@ class EhcacheActiveEntity implements ActiveServerEntity<EhcacheEntityMessage, Eh
 
     final Set<ClientDescriptor> clients = storeClientMap.get(name);
     if (clients != null && !clients.isEmpty()) {
-      throw new ResourceBusyException("Can not destroy clustered tier '" + name + "': in use by " + clients.size() + " clients");
+      throw new ResourceBusyException("Cannot destroy clustered tier '" + name + "': in use by " + clients.size() + " other client(s)");
     }
 
     LOGGER.info("Client {} destroying clustered tier '{}'", clientDescriptor, name);

